### PR TITLE
rudimentary matching decompilation of NitroMain

### PR DESCRIPTION
**Pull request by Demki**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+extern const s32 CONST_3F; // 0x0000003F
+extern const s32 CONST_34; // 0x00000034
+
+extern struct Unk21DBE18 gUnk021DBE18; 
+extern struct Unk21DBE18 gUnk021D76C8;
+
+
+void NitroMain(void)
+{
+	FUN_02016014();
```
Comment (ProjectRevoTPP) [inline src/main.c:70]: tabs to spaces (tab = 4 spaces)

Comment (red031000) [reply src/main.c:70]: can we get an editorconfig to enforce that pls?

File: asm/FUN_020C9C0C.s [hunk 2]
```diff
@@ -0,0 +1,3098 @@
...
+
+	.text
+
+	arm_func_start OS_WaitIrq
+OS_WaitIrq: ; 0x020C9C0C
+	stmdb sp!, {r4-r6,lr}
+	mov r5, r0
+	mov r4, r1
+	bl FUN_020CD310
+_020C9C1C:
```
Comment (PikalaxALT) [inline asm/FUN_020C9C0C.s:14]: disassemble remainder

Comment (Demki) [reply asm/FUN_020C9C0C.s]: done

File: global.inc [hunk 3]
```diff
@@ -905,4 +905,25 @@
...
+.extern PMi_SetLEDAsync
+.extern PMi_ReadRegister
+.extern PMi_WaitBusy
+.extern CTRDG_IsExisting
+.extern PMi_SendSleepStart
+.extern FUN_01FF857C
+.extern FUN_01FF84E4
+.extern FUN_01FF85F0
+.extern FUN_01FF84A4
+.extern FUN_01FF8544
```
Comment (PikalaxALT) [inline global.inc:929]: Missing newline at EOF

File: asm/secure.s [hunk 4]
```diff
@@ -2,4 +2,12 @@
 
 .section .text
 
-.incbin "baserom.nds", 0x4000, 0x800
+.incbin "baserom.nds", 0x4000, 0x5F2
+
+	.global FUN_020005F2
```
Comment (PikalaxALT) [inline asm/secure.s:7]: `thumb_func_start`

**Pull request merged**